ca/external.go

```diff
@@ -154,7 +155,8 @@ func (eca *ExternalCA) CrossSignRootCA(ctx context.Context, rca RootCA) ([]byte,
 }
 
 func makeExternalSignRequest(ctx context.Context, client *http.Client, url string, csrJSON []byte) (cert []byte, err error) {
-	resp, err := ctxhttp.Post(ctx, client, url, "application/json", bytes.NewReader(csrJSON))
+	requestCtx, _ := context.WithTimeout(ctx, 5*time.Second)
```

Comment:
Call the cancel function after the context is no longer needed. That will ensure resources get freed, instead of possibly having them tied up for 5 seconds.